### PR TITLE
Rewrite JacobiSNCNDNReduced

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: numerics/elliptic_functions.cpp [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
+
+using quantities::Sqrt;
+
 namespace numerics {
 namespace {
 
 void JacobiSNCNDNReduced(double u, double mc, double& s, double& c, double& d);
 
+// Maclaurin series for Fukushima b₀.  These are polynomials in m that are used
+// to build a polynomial in u₀².  The index gives the degree of the polynomial.
```
Comment (eggrobin) [inline numerics/elliptic_functions.cpp]: > that are used ~to build~ *as coefficients* of a polynomial in u₀²

> The index gives the degree of the polynomial

The index should give the corresponding power of u₀² .

File: numerics/elliptic_functions.cpp [hunk 2]
```diff
@@ -32,70 +52,64 @@ void JacobiSNCNDNReduced(double const u,
...
+
+  // We use the subscript i to indicate variables that are computed as part of
+  // the iteration (Fukushima uses subscripts n and N).  This avoids confusion
+  // between c (the result) and cᵢ (the intermediate numerator of c).
+  double bᵢ = fukushima_b₀_maclaurin_u₀²_3.Evaluate(u₀²);
+
+  double const uA = 1.76269 + 1.16357 * mc;
+  bool const may_have_cancellation = u > uA;
+  double aᵢ = 1.0;
+  for (int j = 0; j < n; ++j) {
```
Comment (eggrobin) [inline numerics/elliptic_functions.cpp]: Rename `j` to `i`, for consistency with the subscript.

File: numerics/elliptic_functions.cpp [hunk 3]
```diff
@@ -32,70 +52,64 @@ void JacobiSNCNDNReduced(double const u,
...
+    double const yᵢ = bᵢ * (2.0 * aᵢ - bᵢ);
+    double const zᵢ = aᵢ * aᵢ;
+    double const myᵢ = m * yᵢ;
+    if (may_have_cancellation && zᵢ < 2.0 * myᵢ) {
+      double cᵢ = aᵢ - bᵢ;
+      double const two_mc = 2.0 * mc;
+      double const two_m = 2.0 * m;
+      for (int i = j; i < n; ++i) {
```
Comment (eggrobin) [inline numerics/elliptic_functions.cpp]: This just finishes the outer loop; reuse the same index accordingly.

File: numerics/elliptic_functions.cpp (hunk 1 above)
Comment (pleroy) [reply numerics/elliptic_functions.cpp]: Done.

File: numerics/elliptic_functions.cpp (hunk 2 above)
Comment (pleroy) [reply numerics/elliptic_functions.cpp]: Done.

File: numerics/elliptic_functions.cpp (hunk 3 above)
Comment (pleroy) [reply numerics/elliptic_functions.cpp]: Done.

**Pull request merged**